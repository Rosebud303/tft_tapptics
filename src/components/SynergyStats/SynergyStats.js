import React, { Component } from 'react';
import ReactTooltip from 'react-tooltip';


export class SynergyStats extends Component {

  // const compareSynergies = arr => {
  //   // iterate over roster
  //   let currentSynergies = arr.reduce((acc, champ) => {
  //   // iterate over champ.origin_class_types
  //     champ.attributes.origin_class_types.forEach(type => {
  //       if (type && !acc[type.name]) {
  //         acc[type.name] = 0
  //       }
  //   // add to type.name value on obj
  //       acc[type.name]++
  //     })
  //     return acc
  //   }, {})
  //   console.log(currentSynergies)

  //   let bonuses = synergies.reduce((acc, syn) => {
  //     if (syn.attributes.tiers.includes(currentSynergies[syn.attributes.name])) {
  //       acc.push(currentSynergies[syn.attributes.name])
  //     }
  //     return acc
  //   }, [])
  //   console.log(bonuses)
  //   // if type.tiers includes value,
  //     // get index
  //     // return tier_info at index into p tag
  // }

  render () {

    let currentSynergies = this.props.roster.reduce((acc, champ) => {
      // iterate over champ.origin_class_types
        champ.attributes.origin_class_types.forEach(type => {
          if (type && !acc[type.name]) {
            acc[type.name] = 0
          }
      // add to type.name value on obj
          acc[type.name]++
        })
        return acc
      }, {})
      console.log(currentSynergies)
  
      let bonuses = this.props.synergies.reduce((acc, syn) => {
        if (syn.attributes.tiers.includes(currentSynergies[syn.attributes.name])) {
          let index = syn.attributes.tiers.indexOf(currentSynergies[syn.attributes.name])
          acc.push(syn.attributes.tier_info[index])
        }
        return acc
      }, [])
      console.log(bonuses)

    return (
  
      
      <article className="SynergyStats">
      {/* {compareSynergies(this.props.roster)} */}
        {/* <a data-tip data-for={thumbnail}>
          <img src={thumbnail} alt={name} />
        </a>
        <ReactTooltip id={thumbnail} type='dark' effect="float" className="modal-tooltip">
          <h6>{name}</h6>
        </ReactTooltip> */}
        {bonuses}
      </article>
    )
  }
}

export default SynergyStats;



// assassin: 0,
//       blademaster: 0,
//       brawler: 0,
//       elementalist: 0,
//       guardian: 0,
//       gunslinger: 0,
//       knight: 0,
//       ranger: 0,
//       shapeshifter: 0,
//       sorcerer: 0,
//       demon: 0,
//       dragon: 0,
//       exile: 0,
//       glacial: 0,
//       robot: 0,
//       imperial: 0,
//       noble: 0,
//       ninja: 0,
//       pirate: 0,
//       phantom: 0,
//       wild: 0,
//       void: 0,
//       yordle: 0