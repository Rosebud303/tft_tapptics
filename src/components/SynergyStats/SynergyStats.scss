.SynergyStats {
  margin: .5rem auto;

  h6, img {
    display: inline;
  }

  h6 {
    font-size: .8rem;
    margin: auto;
  }
}