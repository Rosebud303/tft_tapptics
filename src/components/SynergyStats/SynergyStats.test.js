import React from 'react';
import { shallow } from 'enzyme';
import SynergyStats from './SynergyStats';

describe('SynergyStats', () => {

  let wrapper;
  let name = 'Tristana';
  let thumbnail = 'www.fakeurl.com';
  let roster = [{attributes: {name: 'Nidalee', origin_class_types: ['Wild'] }}]
  let synergies = [{attributes: {name: 'wild', tiers: [2, 4], tier_info: ['50% chance to gain awesomeness', '2% chance to gain eternal life'] }}]

  beforeEach(() => {
    wrapper = shallow(<SynergyStats name={name} thumbnail={thumbnail} roster={roster} synergies={synergies} />)
  });

  it('should match the snapshot with the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });
});