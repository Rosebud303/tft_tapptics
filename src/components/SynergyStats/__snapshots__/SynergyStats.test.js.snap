// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SynergyStats should match the snapshot with the data passed in 1`] = `
<article
  className="SynergyStats"
/>
`;
